lib/active_merchant/billing/gateways/openpay.rb

```diff
@@ -21,9 +28,20 @@ def initialize(options = {})
         requires!(options, :key, :merchant_id)
         @api_key = options[:key]
         @merchant_id = options[:merchant_id]
+        assign_urls(options)
         super
       end
 
+      def assign_urls(options = {})
+        self.test_url = options[:merchant_country] == 'MX' ? self.mx_test_url : self.co_test_url
+        self.live_url = options[:merchant_country] == 'MX' ? self.mx_live_url : self.co_live_url
+      end
```

Comment:
class attribute generates accesor for class and instance https://apidock.com/rails/Class/class_attribute

```
## Instances may overwrite the class value in the same way:

Base.setting = true
object = Base.new
object.setting          # => true
object.setting = false
object.setting          # => false
Base.setting            # => true
```